src/test/groovy/org/sonarqube/gradle/AndroidUtilsTest.java

```diff
@@ -120,12 +123,22 @@ void configureForAndroid_androidPropertiesAreSetForAndroidProject_whenVariantDet
     when(baseVariant.getMergedFlavor()).thenReturn(mergedFlavor);
     when(baseVariant.getName()).thenReturn("myVariant");
 
+    // Set up a sourceSet with a file to test the presence of sonar.sources
+    SourceProvider javaSource = mock(SourceProvider.class);
+    File file = mock(File.class);
+    when(file.exists()).thenReturn(true);
+    when(file.getAbsolutePath()).thenReturn("src/main/java/manifest.xml");
+    when(javaSource.getManifestFile()).thenReturn(file);
+    when(baseVariant.getSourceSets()).thenReturn(List.of(javaSource));
+
     Map<String, Object> resultProperties = new HashMap<>();
     AndroidUtils.configureForAndroid(project, "myVariant", resultProperties);
 
     assertEquals(true, resultProperties.get("sonar.android.detected"));
     assertEquals(30, resultProperties.get("sonar.android.minsdkversion.min"));
     assertEquals(30, resultProperties.get("sonar.android.minsdkversion.max"));
+    Set<File> sources = (Set<File>) resultProperties.get("sonar.sources");
+    assertTrue(sources.contains(file));
```

Comment:
We may want to test the size of the sources set just to detect regressions in the future.
```suggestion
    assertTrue(sources.size() == 1);
    assertTrue(sources.contains(file));
```